examples/asr/speech_to_text_finetune.py

```diff
@@ -0,0 +1,137 @@
+initializations:
+1) `init_from_nemo_model`, and
+2) `init_from_pretrained_model`,
+but not `init_from_ptl_ckpt`.
+
+To train with prior base model tokenizer keep `model.tokenizer.update_tokenizer` as false else
+make it true and provide tokenizer dir along with tokenizer type.
+
+To fine-tune the model, use the following commands:
+
+For initialization from a NEMO model:
+```sh
+python <NEMO_ROOT>/examples/asr/speech_to_text_finetune.py \
+    init_from_nemo_model=<path_to_nemo_model>
+```
+
+For initialization from a pretrained model:
+```sh
+python <NEMO_ROOT>/examples/asr/speech_to_text_finetune.py \
+    init_from_pretrained_model=<pretrained_model_name>
+```
+
+# Fine-Tune a Model
+
+For documentation on fine-tuning this model, please visit:
+https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/main/asr/configs.html#fine-tuning-configurations
+"""
+
+import copy
+
+import pytorch_lightning as pl
+from omegaconf import OmegaConf
+from pytorch_lightning.utilities import rank_zero_only
+
+from nemo.collections.asr.models import ASRModel
+from nemo.core.config import hydra_runner
+from nemo.utils import logging, model_utils
+from nemo.utils.exp_manager import exp_manager
+
+
+@hydra_runner(config_path="conf", config_name="speech_to_text_finetune")
+def main(cfg):
+    logging.info(f'Hydra config: {OmegaConf.to_yaml(cfg)}')
+
+    trainer = pl.Trainer(**cfg.trainer)
+    exp_manager(trainer, cfg.get("exp_manager", None))
+
+    if hasattr(cfg, 'init_from_ptl_ckpt') and cfg.init_from_ptl_ckpt is not None:
+        raise NotImplementedError(
+            "Currently for simplicity of single script for all model types, we only support `init_from_nemo_model` and `init_from_pretrained_model`"
+        )
+
+    @rank_zero_only
+    def get_base_model(cfg):
+        asr_model = None
+        nemo_model_path = cfg.get('init_from_nemo_model', None)
+        pretrained_name = cfg.get('init_from_pretrained_model', None)
+        if nemo_model_path is not None and pretrained_name is not None:
+            raise ValueError("Only pass `init_from_nemo_model` or `init_from_pretrained_model` but not both")
+        elif nemo_model_path is None and pretrained_name is None:
+            raise ValueError(
+                "Both `init_from_nemo_model` and `init_from_pretrained_model cannot be None, should pass atleast one of them"
+            )
+        elif nemo_model_path is not None:
+            asr_model = ASRModel.restore_from(restore_path=nemo_model_path)
+        elif pretrained_name is not None:
+            asr_model = ASRModel.from_pretrained(model_name=pretrained_name)
+
+        return asr_model
+
+    asr_model = get_base_model(cfg)
+    vocab_size = asr_model.tokenizer.vocab_size
+
+    # if new tokenizer is provided, use it
+    if hasattr(cfg.model.tokenizer, 'update_tokenizer') and cfg.model.tokenizer.update_tokenizer:
+        decoder = copy.deepcopy(asr_model.decoder)
```

Comment:
Do NOT deep copy pytorch modules. Only call module.state_dict() and use load_state_dict() latef. 

Our modules are not safe to deep copy